File: app/src/main/java/com/summer/demo/dialog/TipDialog.java
package com.summer.demo.dialog;


import android.annotation.SuppressLint;
import android.app.Dialog;
import android.content.Context;
import android.content.DialogInterface;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.TextView;

import com.summer.demo.R;
import com.summer.helper.utils.SUtils;

/**
 * @author xiastars@vip.qq.com
 */
public class TipDialog extends Dialog {

    private DialogAfterClickListener listener;
    private String name;
    private int layoutid;
    private String mLoadUrl;

    TextView tvOK;
    TextView tvCancel;

    public TipDialog(Context context, String name, DialogAfterClickListener listener) {
        super(context, R.style.TagFullScreenDialog);
        this.listener = listener;
        this.name = name;
        layoutid = R.layout.dialog_tips;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(layoutid);

        initView();
    }

    @SuppressLint("SetJavaScriptEnabled")
    private void initView() {
        TextView tips_tv = (TextView) findViewById(R.id.tips_tv);
        if (!TextUtils.isEmpty(name)) {
            tips_tv.setText(name);
        }
        tvOK = (TextView) findViewById(R.id.tips_ok_tv);
        SUtils.clickTransColor(tvOK);
        tvOK.setOnClickListener(new View.OnClickListener() {

            @Override
            public void onClick(View v) {
                if (null != listener) {
                    listener.onSure();
                }
                TipDialog.this.cancel();
            }

        });
        tvCancel = (TextView) findViewById(R.id.tips_cancel_tv);
        SUtils.clickTransColor(tvCancel);
        tvCancel.setOnClickListener(new View.OnClickListener() {

            @Override
            public void onClick(View v) {
                if (null != listener) {
                    listener.onCancel();
                }
                TipDialog.this.cancel();
            }
        });

        setOnCancelListener(new OnCancelListener() {

            @Override
            public void onCancel(DialogInterface dialog) {
                if (null != listener) {
                    listener.onCancel();
                }
            }
        });

    }

    /**
     * 自定义左边按钮的文字
     *
     * @param id
     */
    public void setOkText(int id) {
        tvOK.setText(getContext().getResources().getString(id));
    }

    /**
     * 自定义右边按钮的文字
     *
     * @param id
     */
    public void setCancelText(int id) {
        tvCancel.setText(getContext().getResources().getString(id));
    }

    public interface DialogAfterClickListener {
        void onSure();

        void onCancel();
    }

}

File: app/src/main/java/com/summer/demo/dialog/TipDialog.kt
package com.summer.demo.dialog

import android.annotation.SuppressLint
import android.app.Dialog
import android.content.Context
import android.os.Bundle
import android.text.TextUtils
import android.view.View
import android.widget.TextView
import com.summer.demo.R
import com.summer.helper.utils.SUtils

/**
 * @author xiastars@vip.qq.com
 */
class TipDialog(context: Context?, private val name: String, private val listener: DialogAfterClickListener?) : Dialog(context, R.style.TagFullScreenDialog) {
    private val layoutid: Int
    private val mLoadUrl: String? = null
    var tvOK: TextView? = null
    var tvCancel: TextView? = null
    override fun onCreate(savedInstanceState: Bundle) {
        super.onCreate(savedInstanceState)
        setContentView(layoutid)
        initView()
    }

    @SuppressLint("SetJavaScriptEnabled")
    private fun initView() {
        val tips_tv = findViewById<View>(R.id.tips_tv) as TextView
        if (!TextUtils.isEmpty(name)) {
            tips_tv.text = name
        }
        tvOK = findViewById<View>(R.id.tips_ok_tv) as TextView
        SUtils.clickTransColor(tvOK)
        tvOK!!.setOnClickListener {
            listener?.onSure()
            cancel()
        }
        tvCancel = findViewById<View>(R.id.tips_cancel_tv) as TextView
        SUtils.clickTransColor(tvCancel)
        tvCancel!!.setOnClickListener {
            listener?.onCancel()
            cancel()
        }
        setOnCancelListener { listener?.onCancel() }
    }

    /**
     * 自定义左边按钮的文字
     *
     * @param id
     */
    fun setOkText(id: Int) {
        tvOK!!.text = context.resources.getString(id)
    }

    /**
     * 自定义右边按钮的文字
     *
     * @param id
     */
    fun setCancelText(id: Int) {
        tvCancel!!.text = context.resources.getString(id)
    }

    interface DialogAfterClickListener {
        fun onSure()
        fun onCancel()
    }

    init {
        layoutid = R.layout.dialog_tips
    }
}